src/AltinnCore/Designer/Controllers/RuntimeAPIController.cs

```diff
@@ -232,25 +233,88 @@ public FileResult GetInstanceFromFile(string org, string service, string develop
         /// <param name="partyId">The party id of the test user</param>
         /// <param name="instanceId">The instance id</param>
         /// <param name="attachmentType">The attachment type id</param>
-        /// <param name="fileExtension">The name of the attachment</param>
-        /// <returns>The status of the upload</returns>
+        /// <param name="attachmentName">The name of the attachment</param>
+        /// <returns>The status of the upload and guid of attachment</returns>
         [HttpPost]
         [DisableFormValueModelBinding]
-        public async System.Threading.Tasks.Task<IActionResult> SaveFormAttachment(string org, string service, string developer, int partyId, int instanceId, string attachmentType, string fileExtension)
+        public async System.Threading.Tasks.Task<IActionResult> SaveFormAttachment(string org, string service, string developer, int partyId, Guid instanceId, string attachmentType, string attachmentName)
         {
             Guid guid = Guid.NewGuid();
-            string pathToSaveTo = _settings.GetTestdataForPartyPath(org, service, developer) + "{0}/{1}/data/{2}/{3}";
-            Directory.CreateDirectory(string.Format(pathToSaveTo, partyId, instanceId, attachmentType, string.Empty));
-            string fileToWriteTo = string.Format(pathToSaveTo, partyId, instanceId, attachmentType, guid.ToString() + fileExtension);
+            string pathToSaveTo = _settings.GetTestdataForPartyPath(org, service, developer) + "{0}/{1}/data/{2}/{3}/{4}";
+            Directory.CreateDirectory(string.Format(pathToSaveTo, partyId, instanceId, attachmentType, guid, string.Empty));
+            string fileToWriteTo = string.Format(pathToSaveTo, partyId, instanceId, attachmentType, guid, attachmentName);
```

Comment:
I prefer using $"{_settings.GetTestdataForPartyPath(org, service, developer)}{partyId}/{instanceId}/data/{attachmentType}/{guid}" over string.format. But I see it was already like this to begin with, but for the other two methods you can consider if you want to change it or not

Reply:
I agree, makes it a lot more readable. Changed for all three functions.